### PR TITLE
Experience/13578/api elims fix using feature flag

### LINKED ISSUE #13578: [BUG] Temporary fix to Daily Data API to include ELIMS files
## Problem Statement
The Daily Data page does not show ELIMS data.

## What you need to know
- ELIMS data passthrough does not go through the batch step, which is the step that Daily Data API queries.
- The temporary fix involves changing the call to something like this:
`OR ("public"."action"."action_name" = cast('send' as "public"."task_action")
and "public"."report_file"."schema_topic" = 'elr-elims'))`
- For questions, go to Victor

## Acceptance criteria
- [x] All ELIMS sent items appear in the Daily Data table alongside other data in accordance with existing logic
  - [x] ELIMS Daily Data Page in Staging
  - [x] NY-PHD Daily Data Page in Staging
  - [x] CA-PHD Daily Data Page in Staging
- [ ] Spotcheck 4-5 receivers Daily Data pages that are NOT receiving ELIMs data to ensure they are not affected by this change

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Fix to Daily Data API to include ELIMS files

Test Steps:

Log into RS
All ELIMS sent items appear in the Daily Data table alongside other data in accordance with existing logic

## Changes
Added additional logic to the organizationFilter method which cjecks for feature flag on organization

Query if org DOES NOT HAVE feature flag set (original logic)
```
select
    "public"."action"."action_id",
    "public"."action"."created_at",
    "public"."action"."sending_org",
    "public"."action"."sending_org_client",
    "public"."report_file"."receiving_org",
    "public"."report_file"."receiving_org_svc",
    ("public"."setting"."values"->>'customerStatus') as "receiving_org_svc_status",
    "public"."action"."http_status",
    "public"."action"."external_name",
    "public"."report_file"."report_id",
    "public"."report_file"."schema_topic",
    "public"."report_file"."item_count",
    "public"."report_file"."body_url",
    "public"."report_file"."schema_name",
    "public"."report_file"."body_format"
from "public"."action"
         join "public"."report_file"
              on "public"."report_file"."action_id" = "public"."action"."action_id"
         join "public"."setting"
              on (
                  "public"."report_file"."receiving_org" = ("public"."setting"."values"->>'organizationName')
                      and "public"."report_file"."receiving_org_svc" = ("public"."setting"."values"->>'name')
                      and "public"."setting"."is_active"
                  )
where (
          "public"."action"."action_name" = cast('batch' as "public"."task_action")
              and "public"."report_file"."receiving_org" = 'ak-phd'
              and "public"."report_file"."receiving_org_svc" = 'elr'
          )
order by "public"."action"."created_at" desc
```
Query if org does HAS feature flag set
```
select
    "public"."action"."action_id",
    "public"."action"."created_at",
    "public"."action"."sending_org",
    "public"."action"."sending_org_client",
    "public"."report_file"."receiving_org",
    "public"."report_file"."receiving_org_svc",
    ("public"."setting"."values"->>'customerStatus') as "receiving_org_svc_status",
    "public"."action"."http_status",
    "public"."action"."external_name",
    "public"."report_file"."report_id",
    "public"."report_file"."schema_topic",
    "public"."report_file"."item_count",
    "public"."report_file"."body_url",
    "public"."report_file"."schema_name",
    "public"."report_file"."body_format"
from "public"."action"
         join "public"."report_file"
              on "public"."report_file"."action_id" = "public"."action"."action_id"
         join "public"."setting"
              on (
                  "public"."report_file"."receiving_org" = ("public"."setting"."values"->>'organizationName')
                      and "public"."report_file"."receiving_org_svc" = ("public"."setting"."values"->>'name')
                      and "public"."setting"."is_active"
                  )
where (
          (
              "public"."action"."action_name" = cast('batch' as "public"."task_action")
                  or (
                  "public"."action"."action_name" = cast('send' as "public"."task_action")
                      and "public"."report_file"."schema_topic" = 'ELR_ELIMS'
                  )
              )
              and "public"."report_file"."receiving_org" = 'ignore'
              and "public"."report_file"."receiving_org_svc" = 'FULL_ELR'
          )
order by "public"."action"."created_at" desc
```

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

## Linked Issues
- Fixes #13578 

## To Be Done
- Unit tests

### CODE REVIEW

**Review by victor-chaparro — APPROVED** [review]
LGTM

File: prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt [hunk 1]
```diff
@@ -41,8 +42,17 @@ class DatabaseDeliveryAccess(
...
         orgService: String?,
     ): Condition {
-        var filter = ACTION.ACTION_NAME.eq(TaskAction.batch)
-            .and(REPORT_FILE.RECEIVING_ORG.eq(organization))
+//      This is a temporary fix to make the daily data page show data for reports routed through the ELR ELIMS topic since reports routed through the ELR-ELIMS topic skip the batch step. There are plans to have a dashboard for receivers that works for all messages routed through RS.
+        val org = BaseEngine.settingsProviderSingleton.findOrganization(organization)
+
+        var filter = if (org?.featureFlags?.contains("ELIMS_DATA") == true) {
+                ACTION.ACTION_NAME.eq(TaskAction.batch)
+                    .or(ACTION.ACTION_NAME.eq(TaskAction.send).and(REPORT_FILE.SCHEMA_TOPIC.eq(Topic.ELR_ELIMS)))
```
Comment (mkalish) [inline prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt:50]: I would maybe run these in production in metabase?  It doesn't look like report_file.schema_topic is indexed

**Review by mkalish — APPROVED** [review]
Changes look fine to me.  Only comment is to just check in prod for one the big receivers for timing on the query

File: prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt (hunk 1 above)
Comment (victor-chaparro) [reply prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt:50]: @mkalish Here's a PR showing the explain plan in production before and after we added the receiving org and service index. It doesn't look like it needs an index on the schema_topic.

**Pull request merged**